src/routes/api/tasks/index.ts

```diff
@@ -299,6 +303,50 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
       )
     }
   )
+
+  fastify.delete(
+    '/:filename/image',
+    {
+      schema: {
+        params: Type.Object({
+          filename: Type.String()
+        }),
+        response: {
+          204: Type.Null(),
+          404: Type.Object({ message: Type.String() })
+        },
+        tags: ['Tasks']
+      }
+    },
+    async function (request, reply) {
+      const { filename } = request.params
+
+      return fastify.knex.transaction(async (trx) => {
+        const numberOfRowUpdated = await trx<Task>('tasks')
+          .where({ filename })
+          .update({ filename: null })
+
+        if (numberOfRowUpdated === 0) {
+          return reply.notFound(`No task has filename "${filename}"`)
+        }
+
+
+        const filePath = path.join(
+          import.meta.dirname,
+          '../../../..',
+          fastify.config.UPLOAD_DIRNAME,
+          fastify.config.UPLOAD_TASKS_DIRNAME,
+          filename
+        )
+
+        await fs.promises.unlink(filePath)
```

Comment:
If `ENOENT` error, we should return 404 Not found in the `catch` statement.

Reply:
Then i have to mock the unlink function for the rollback test, no ?

Reply:
Not for 404 error if file is not found. But yes for throwing an error that will lead to 500 internal server error.

Reply:
I always prefer consider it as success even no file is deleted.
The reason is very simple, file system can be manged outside of the system. It is easy to have mismatch in file system and database when the system admin did something wrong. I consider when database operation is success, it is always success (no matter the result of file system operation).

This only valid for removing file, update or create should requires both operation to success.

Reply:
So i should catch the unlink error and just send a 204 response ?

Reply:
That's only my personal preference, raised up for reference only.
Not necessary to change.

Reply:
Yep, good point.


Maybe we can still log the error in case of a real issue?
```ts
await fs.promises.unlink('filePath').catch(() => {
  fastify.log.warn(`File path '${filePath}' not found`)
});
```

Reply:
> That's only my personal preference,

I think this is objectively the right approach.

Reply:
Can you do the change @Turn0xx, then I merge.

Reply:
Yes, i gonna push this morning 👌

Reply:
I would add a comment that sums up Climba's point, this demo is meant to be pedagogic.